homeassistant/components/xiaomi_miio/__init__.py

```diff
@@ -397,8 +425,23 @@ async def async_setup_gateway_entry(hass: HomeAssistant, entry: ConfigEntry) ->
         raise ConfigEntryNotReady() from error
     gateway_info = gateway.gateway_info
 
+    def event_callback_factory(device_id):
+        """Create event callback for a subdevice."""
+
+        @callback
+        def event_callback(action, params):
+            """Event from subdevice."""
+            _LOGGER.debug("Got new event_callback: %s, %s", action, params)
+            hass.bus.fire(
```

Comment:
Use the async api in async context.
```suggestion
            hass.bus.async_fire(
```